incubator/java-spring-boot2/image/Dockerfile-stack

```diff
@@ -11,30 +11,27 @@ RUN  apt-get -qq update \
 COPY ./project /project
 COPY ./mvn-stack-settings.xml /usr/share/maven/conf/settings.xml
 COPY ./mvn-stack-settings.xml /project/.mvn-stack-settings.xml
-COPY ./java-spring-boot2-build.sh /project/java-spring-boot2-build.sh
 
-WORKDIR "/project"
-
-RUN mkdir -p /project/target && chmod 777 /project/target
+WORKDIR /project/user-app
 
 ENV APPSODY_USER_RUN_AS_LOCAL=true
 
-ENV APPSODY_MOUNTS="src:/project/src;pom.xml:/project/pom.xml;~/.m2/repository:/mvn/repository"
+ENV APPSODY_MOUNTS=".:/project/user-app;~/.m2/repository:/mvn/repository"
```

Comment:
Interestingly the other stack( java-microprofile) went from mounting a single workspace dir to mounting src and pom individually as there were issues with creation of file / folders  that were in the directory that was bind mounted on the host. Were there specific reasons for changing the way the folders are mounted?

Reply:
This is the most significant change, and is what allows the version plugin to run correctly. Remember that the Spring stack runs as user, so any files we create have proper ownership.  I’m not sure the microprofile stack does that yet.. that may have been a different way to fix the issue they were having.

Reply:
Yes, run as user is one of the fundamental differences between the 2 stacks.. and maybe it is good to have two different approaches so we can compare and contrast experiences from both before going full on on one.